tests/ui/issues/allocator-shim-circular-deps.rs

```diff
@@ -0,0 +1,20 @@
+// This test is designed to intentionally introduce a circular dependency scenario to check
+// that a specific compiler bug doesn't make a resurgence.
+// The bug in question arose when at least one crate
+// required a global allocator, and that crate was placed after
+// the one defining it in the linker order.
+// The generated symbols.o should not result in any linker errors.
+// See https://github.com/rust-lang/rust/issues/112715
```

Comment:
Suggestion: actually, if this test and its auxillaries are sensitive to link order, let's keep it as a run-make test so we can explicitly control the link order.